Use terminal width for CLI wrapping

tofumatt opened the pull request:
I was using a larger-than-usual terminal window and noticed the output was still wrapped to 78 characters.

This uses process.stdout.columns if available to set the CLI wrap value. (Minus two pixels to preserve a similar padding effect as using 78 over 80).

Kind of a superfluous tweak, but I added a bunch of tests for it and it's nice to be a good CLI citizen like this.

#### Before:
<img width="843" alt="screenshot 2015-10-14 19 59 40" src="https://cloud.githubusercontent.com/assets/90871/10495198/8b871260-72b2-11e5-9aec-2ea6e7e84537.png">

#### After:
<img width="843" alt="screenshot 2015-10-14 20 16 53" src="https://cloud.githubusercontent.com/assets/90871/10495207/96bf9c24-72b2-11e5-8413-7662c8c1d175.png">

muffinresearch commented:
You might want to check the text output too. I glossed over trying to make it work in small terminals because it got cramped fast. But if you can work out a solution there too that fits in with this that would be nice too.

tofumatt commented:
Oh, I figured they were rarely under 78. But I'll make the minumum size lower than 78. 10 should be reasonable.

- tofumatt (Sent from mobile)

> On 14 Oct 2015, at 21:31, Stuart Colville <notifications@github.com> wrote:
> 
> You might want to check the text output too. I glossed over trying to make it work in small terminals because it got cramped fast. But if you can work out a solution there too that fits in with this that would be nice too.
> 
> —
> Reply to this email directly or view it on GitHub.
>

muffinresearch commented:
We might be at cross-purposes I'm talking about the valdation error / warning text output.

tofumatt commented:
Oh my you’re right, that is weird.

I’ll check this out more and push when it’s all good, thanks!

muffinresearch commented:
We can always deal with that in a separate bug - it's all good :)

tofumatt commented:
Yeah, looks like it’s always been that way.

I fixed it here; now the validator output is dependant on the size of the terminal, so it looks better in smaller terminals.

muffinresearch commented:
One thing is the error-codes if the colorization is applied and the heading width is too narrow the colour will bleed into the rest of the content. I filed a bug with columnify [1] upstream to deal with this - but until then we'd need to watch that or just take the colours out.

https://github.com/timoxley/columnify/issues/26

tofumatt commented:
Yeah, I set the heading width to always be 35 regardless of window size, so that shouldn’t happen (in my testing it didn’t, we could write a test for it though combing through those codes to test is a PITA for an upstream bug).

muffinresearch commented:
r+wc

tofumatt commented:
I added a few more tests and removed certain columns in smaller terminals. I added tests for that as well as for terminals as small as 1 column. Tested it in my terminal too, everything seems better now.

muffinresearch commented on tests/test.validator.js:
```diff
@@ -393,11 +412,61 @@ describe('Validator.textOutput()', function() {
+    assert.notInclude(text, '20');
+  });
+
+  it('should survive even a 1 column terminal', () => {
```
That's a niche use-case :)

tofumatt replied:
Indeed, but now we know it won’t crash! :)

muffinresearch commented:
Works as advertised. r+
